modules/http/src/test/java/org/mule/module/http/functional/requester/crl/HttpPreferringCrlTestCase.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.http.functional.requester.crl;
+
+import static org.apache.commons.lang.exception.ExceptionUtils.getRootCause;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import org.mule.module.http.functional.AbstractHttpTlsRevocationTestCase;
+
+import java.security.cert.CertPathValidatorException;
+
+import org.junit.Test;
+
+public class HttpPreferringCrlTestCase extends AbstractHttpTlsRevocationTestCase
+{
+
+    public static String EXPECTED_OCSP_ERROR_MESSAGE = "Certificate does not specify OCSP responder";
+
+    public HttpPreferringCrlTestCase()
+    {
+        super("http-requester-tls-crl-standard-config.xml", REVOKED_CRL_FILE_PATH, ENTITY_CERTIFIED_REVOCATION_SUB_PATH);
+    }
+
+    @Test
+    public void testPreferCrlWithFallback() throws Exception
+    {
+        try
+        {
+            runFlow("testFlowPreferCrl");
+            fail("CertificateRevokedException should have been thrown.");
+        }
+        catch (Exception e)
+        {
+            verifyRevocationException(e);
+        }
+    }
+
```

Comment:
Why the spaces? There's places with 1 space (accurate), others with 2 and other 3!